codecarbon/emissions_tracker.py

```diff
@@ -324,16 +383,21 @@ def _get_geo_metadata(self) -> GeoMetadata:
         return GeoMetadata.from_geo_js(self._data_source.geo_js_url)
 
     def _get_cloud_metadata(self) -> CloudMetadata:
-        return CloudMetadata.from_utils()
+        if self._cloud is None:
+            self._cloud = CloudMetadata.from_utils()
+        return self._cloud
 
 
 def track_emissions(
     fn: Callable = None,
     project_name: str = "codecarbon",
     measure_power_secs: int = 15,
+    measure_occurrence_before_calling_api: int = 2,
```

Comment:
currently unused, shouldn't it be passed down to the initialization of the emission trackers?

Reply:
You're right !